dace/transformation/interstate/gpu_transform_sdfg.py

```diff
@@ -154,19 +160,42 @@ def can_be_applied(self, graph, expr_index, sdfg, permissive=False):
                     return False
         return True
 
-    def apply(self, _, sdfg: sd.SDFG):
+    def _get_marked_inputs_and_outputs(self, state, entry_node) -> list:
+        if (self.host_data is None or self.host_data == []) and (self.host_maps is None or self.host_maps == []):
+            return []
+        marked_sources = [state.memlet_tree(e).root().edge.src for e in state.in_edges(entry_node)]
+        marked_destinations = [state.memlet_tree(e).root().edge.dst for e in state.in_edges(state.exit_node(entry_node))]
+        marked_accesses = [n.data for n in (marked_sources + marked_destinations) if isinstance(n, nodes.AccessNode) and n.data in self.host_data]
```

Comment:
This will not work with views

Reply:
You mean that it will not work if the src is a view, and then the memlet_tree's root will be a view and the original array?
In this case, I need to check if we have a root that is view and find the original array the view references to and use the original array?

Reply:
yes, I think there is a helper for that called get_all_viewed_nodes or something in sdutil

Reply:
I added some checks for cases with views and extended this to support views